internal/liveshare/test/server.go

```diff
@@ -145,73 +155,145 @@ func makeConnection(server *Server) http.HandlerFunc {
 			// validate the sas key
 			sasParam := req.URL.Query().Get("sb-hc-token")
 			if sasParam != server.relaySAS {
-				server.errCh <- errors.New("error validating sas")
+				sendError(server.errCh, errors.New("error validating sas"))
 				return
 			}
 		}
 		c, err := upgrader.Upgrade(w, req, nil)
 		if err != nil {
-			server.errCh <- fmt.Errorf("error upgrading connection: %w", err)
+			sendError(server.errCh, fmt.Errorf("error upgrading connection: %w", err))
 			return
 		}
-		defer c.Close()
+		defer func() {
+			if err := c.Close(); err != nil {
+				sendError(server.errCh, err)
+			}
+		}()
 
 		socketConn := newSocketConn(c)
 		_, chans, reqs, err := ssh.NewServerConn(socketConn, server.sshConfig)
 		if err != nil {
-			server.errCh <- fmt.Errorf("error creating new ssh conn: %w", err)
+			sendError(server.errCh, fmt.Errorf("error creating new ssh conn: %w", err))
 			return
 		}
 		go ssh.DiscardRequests(reqs)
 
-		for newChannel := range chans {
-			ch, reqs, err := newChannel.Accept()
+		if err := handleChannels(ctx, server, chans); err != nil {
+			sendError(server.errCh, err)
+		}
+	}
+}
+
+func sendError(errc chan<- error, err error) {
+	select {
+	case errc <- err:
+	default:
+	}
+}
+
+func awaitError(ctx context.Context, errc <-chan error) error {
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case err := <-errc:
+		return err
+	}
+}
+
+func handleChannels(ctx context.Context, server *Server, sshChannels <-chan ssh.NewChannel) error {
+	errc := make(chan error, 1)
```

Comment:
`errc` is declared with buffer size 1, but errc might receive an error object for each iteration of the loop over `sshChannels`. Is the buffer size therefore innacurate?